beast/plotting/plot_indiv_fit.py

```diff
@@ -127,7 +126,73 @@ def plot_1dpdf(ax, pdf1d_hdu, tagname, xlabel, starnum, stats=None, logx=False):
         ax.plot(pvals[1:3], [ym, ym], "-", color="m")
 
 
-def plot_beast_ifit(filters, waves, stats, pdf1d_hdu, starnum):
+def plot_indiv_fit(filebase, starnum=0, savefig=False, plotfig=True):
+    """
+    Plot the individual fit for a single observed star including best fit &
+    percentile parameters and various 1D pPDFs
+
+    Parameters
+    ----------
+    filebase : str
+        base filename of run
+
+    starnum : int
+        number of star in the stats file
+
+    savefig : str
+        set to the file extension fo the desired plot file (e.g., png, pdf, etc)
+
+    plotfig : boolean
+        plot the figure to a file or the screen based on savefig
+        otherwise return the fig object
+    """
+
+    starnum = int(starnum)
+
+    # determine how the stats/pdf1d filenames are to be set
+    if len(np.atleast_1d(filebase)) == 1:
+        stats_fname = f"{filebase}_stats.fits"
+        pdf1d_fname = f"{filebase}_pdf1d.fits"
+    else:
+        stats_fname = filebase[0]
+        pdf1d_fname = filebase[1]
+
+    # read in the stats
+    stats = Table.read(stats_fname, hdu=1)
+
+    # check how many extensions the stats file has
+    # determines how to get the filternames and wavelengths
+    with fits.open(stats_fname) as hdul:
+        nhdu = len(hdul)
+    if nhdu > 2:
+        filter_info = Table.read(stats_fname, hdu=2)
+        bfilters = filter_info["filternames"].data
+        waves = filter_info["wavelengths"].data
+        filters = [cfilter.decode("utf-8") for cfilter in bfilters]
+    else:  # PHAT values as default to support old stats files
```

Comment:
Not quite sure what you are asking for here.  I'm sure it is possible.  Maybe open an issue for this with the details and this can be done in a separate PR.

Reply:
Sounds good!